# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/base/nsLayoutUtils.cpp
# Commit: a261bb633790
# Full Hash: a261bb633790ddca9cc8ce83adca0a1bb36114a6
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2022-02-23 09:38:00
# Regressor Bug: 1714584
# File Overlap Count: 2
# Description:
#   Bug 1714584 - Part 1: Decouple nsDisplayList internal list from nsDisplayItems r=mstange
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D138152
# ==============================================================================

diff -r 635f402efb1b -r a261bb633790 layout/base/nsLayoutUtils.cpp
--- a/layout/base/nsLayoutUtils.cpp	Tue Feb 22 23:42:08 2022 +0000
+++ b/layout/base/nsLayoutUtils.cpp	Tue Feb 22 23:42:18 2022 +0000
@@ -2521,7 +2521,7 @@
                                nsDisplayListBuilderMode::TransformComputation,
                                false /*don't build caret*/);
   builder.BeginFrame();
-  nsDisplayList list;
+  nsDisplayList list(&builder);
   nsDisplayTransform* item =
       MakeDisplayItem<nsDisplayTransform>(&builder, aFrame, &list, nsRect());
   MOZ_ASSERT(item);
@@ -2716,7 +2716,7 @@
   nsDisplayListBuilder builder(frame, nsDisplayListBuilderMode::EventDelivery,
                                false);
   builder.BeginFrame();
-  nsDisplayList list;
+  nsDisplayList list(&builder);
 
   if (aOptions.mBits.contains(FrameForPointOption::IgnorePaintSuppression)) {
     builder.IgnorePaintSuppression();
@@ -3058,7 +3058,7 @@
   TemporaryDisplayListBuilder(nsIFrame* aFrame,
                               nsDisplayListBuilderMode aBuilderMode,
                               const bool aBuildCaret)
-      : mBuilder(aFrame, aBuilderMode, aBuildCaret) {}
+      : mBuilder(aFrame, aBuilderMode, aBuildCaret), mList(&mBuilder) {}
 
   ~TemporaryDisplayListBuilder() { mList.DeleteAll(&mBuilder); }
 
@@ -3354,7 +3354,6 @@
         }
 
         list->DeleteAll(builder);
-        list->RestoreState();
 
         builder->ClearRetainedWindowRegions();
         builder->ClearWillChangeBudgets();